src/utils/node.ts

```diff
@@ -29,3 +29,26 @@ export const generateURL = ({ username, password, protocol, host, port, params,
 
   return uri.toString()
 }
+
+export const generateHysteria2URL = ({
+  protocol,
+  auth,
+  host,
+  port,
+  params,
+}: {
+  protocol: string
+  auth: string
+  host: string
+  port: number
+  params: Record<string, string | number | boolean>
+}) => {
+  const uri = new URL(`${protocol}://${auth}@${host}:${port}/`)
```

Comment:
In some cases, the auth field contains the character `@`. We should consider implementing an escape mechanism.

Reply:
Thank you very much for your professional advice, fixed

Reply:
I have an issue where my hysteria2 server cannot connect properly if the password contains @, regardless of whether the client escapes it or not. I'm not sure if it's a problem with the implementation on the dae side or if the hysteria2 server's password settings themselves don't support `@`

server eg:
 ```
 auth:
  type: password
  password: '@99888'
  ```